mesh/twitter_profile_agent.py

```diff
@@ -0,0 +1,496 @@
+import asyncio
+import logging
+import os
+import re
+from typing import Any, Dict, List
+
+import requests
+from dotenv import load_dotenv
+
+from core.llm import call_llm_async
+from decorators import monitor_execution, with_cache, with_retry
+
+from .mesh_agent import MeshAgent
+
+logger = logging.getLogger(__name__)
+load_dotenv()
+
+
+class TwitterProfileAgent(MeshAgent):
+    def __init__(self):
+        super().__init__()
+        self.api_key = os.getenv("APIDANCE_API_KEY")
+        if not self.api_key:
+            raise ValueError("APIDANCE_API_KEY environment variable is required")
+
+        self.twitter_user_api = "https://api.apidance.pro/1.1/users/show.json"
+        self.twitter_tweets_api = "https://api.apidance.pro/sapi/UserTweets"
+        self.headers = {"apikey": self.api_key}
+
+        self.metadata.update(
+            {
+                "name": "Twitter Profile Agent",
+                "version": "1.0.0",
+                "author": "Heurist Team",
+                "author_address": "0x7d9d1821d15B9e0b8Ab98A058361233E255E405D",
+                "description": "This agent fetches a Twitter user's profile information and recent tweets. It's useful for getting project updates or tracking key opinion leaders (KOLs) in the space.",
+                "inputs": [
+                    {
+                        "name": "query",
+                        "description": "Twitter username (with or without @) or user ID to fetch profile and tweets for",
+                        "type": "str",
+                        "required": False,
+                    },
+                    {
+                        "name": "raw_data_only",
+                        "description": "If true, return only raw data without natural language response",
+                        "type": "bool",
+                        "required": False,
+                        "default": False,
+                    },
+                ],
+                "outputs": [
+                    {
+                        "name": "response",
+                        "description": "Natural language summary of the user's profile and recent tweets",
+                        "type": "str",
+                    },
+                    {
+                        "name": "data",
+                        "description": "Structured data containing user profile and recent tweets",
+                        "type": "dict",
+                    },
+                ],
+                "external_apis": ["Twitter API"],
+                "tags": ["Social", "Twitter"],
+                "recommended": True,
+                "image_url": "https://raw.githubusercontent.com/heurist-network/heurist-agent-framework/refs/heads/main/mesh/images/Twitter.png",
+                "examples": [
+                    "Get recent tweets from @username",
```

Comment:
Keep 3 examples
- Summarize recent updates of @heurist_ai
- What has @elonmust been tweeting lately?
- Get recent tweets from @cz_binance